pkg/services/ngalert/eval/eval.go

```diff
@@ -155,11 +162,57 @@ func executeCondition(ctx AlertExecCtx, c *models.Condition, now time.Time, data
 		return ExecutionResults{Error: err}
 	}
 
+	// eval captures for the '__value__' label.
+	captures := []NumberValueCapture{}
+
+	captureVal := func(refID string, labels data.Labels, value *float64) {
+		captures = append(captures, NumberValueCapture{
+			Var:    refID,
+			Value:  value,
+			Labels: labels.Copy(),
```

Comment:
Given we're doing comparisons below and from what I can see its only purpose is to attach them to the actual results, I think it would be very beneficial if we started using the fingerprint of these.

I know this type comes from the go sdk, but we can do something like `fingerprint := model.LabelsToSignature(labels.Copy())` here.

Reply:
Not sure I fully understand.

Saying replace `theseLabels.Equals(cap.Labels) || theseLabels.Contains(cap.Labels) || cap.Labels.Contains(theseLabels)` with a fingerprint, maybe using the Intersection method?

Reply:
I meant replacing the first equality comparison, do we need the union as well? Under which scenarios would a subset of the labels show up on either side?